providers/gcs/gcs.go

```diff
@@ -325,7 +325,7 @@ func (b *Bucket) Handle() *storage.BucketHandle {
 func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
 	if _, err := b.bkt.Object(name).Attrs(ctx); err == nil {
 		return true, nil
-	} else if err != storage.ErrObjectNotExist {
+	} else if b.IsObjNotFoundErr(err) {
```

Comment:
Shouldn't this be the inverse?

```suggestion
	} else if !b.IsObjNotFoundErr(err) {
```

Reply:
🤦 Good catch, fixed.